sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1382,6 +1382,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val STATEFUL_OPERATOR_CORRECTNESS_CHECK_ENABLED =
+    buildConf("spark.sql.streaming.statefulOperator.correctnessCheck")
+      .internal()
+      .doc("When true, the stateful operators for streaming query will be checked for possible " +
+        "correctness issue. Once the issue is detected, Spark will throw analysis exception. " +
```

Comment:
this should have more information about the correctness issue or point to somewhere that does so users can properly make a decision.

Reply:
Added more info. Thanks.